var/spack/repos/builtin/packages/clingo/package.py

```diff
@@ -56,11 +56,35 @@ def cmake_args(self):
         except UnsupportedCompilerFlag:
             InstallError('clingo requires a C++14-compliant C++ compiler')
 
-        return [
-            '-DCLINGO_REQUIRE_PYTHON=ON',
-            '-DCLINGO_BUILD_WITH_PYTHON=ON',
-            '-DCLINGO_BUILD_PY_SHARED=ON',
-            '-DPYCLINGO_USER_INSTALL=OFF',
-            '-DPYCLINGO_USE_INSTALL_PREFIX=ON',
+        spec = self.spec
+        args = [
             '-DCLINGO_BUILD_WITH_LUA=OFF'
         ]
+
+        if '+python' in spec:
+            args += [
+                '-DCLINGO_BUILD_WITH_PYTHON=ON',
+                '-DCLINGO_REQUIRE_PYTHON=ON',
+                '-DCLINGO_BUILD_PY_SHARED=ON',
```

Comment:
What does this flag control? Does it depend on whether `'+shared' in spec['python']`?

Reply:
Good question, unfortunately I'm not that familiar with the internals of the library, maybe the original author of this package can shed some light. There seems to be an additional library `libpyclingo` that uses this flag. As far as I can tell, this library is not used by the Python module, which links against `libclingo.so` directly.

Reply:
Alright, maybe @tgamblin knows